### ISSUE TITLE
TDL Syntax is excessively verbose and hard to read for symbols and s-expressions

### ISSUE THREAD

**Issue by popematt** [amzn/ion-docs#343]
Any macro that uses symbol literals or s-expressions is really verbose and difficult to read. For example, this is a macro that appends macros to the local macro table.
```
(macro append_macros ($macro_definitions*)
    (annotate
        (literal $ion_encoding)
        (make_sexp [
            (literal (symbol_table $ion_encoding)),
            (make_sexp [
                (literal macro_table),
                (literal $ion_encoding),
                $macro_definitions
            ])
        ])
    )
)
```
I have also noticed that early users who are writing macros are falling into a convention of writing variable names with a `$` so that they are more visible.

It would be nice if we could avoid having to escape symbols and s-expressions so that any sort of macro that emits encoding directives or TDL won't be so excessively difficult to read.

**Comment by zslayton** [comment]
If we follow the approach described in #344 (changing TDL invocation syntax from `(foo ...)` to `(.foo ...)`, we could also modify the syntax for variable expansion. Rather than treating all symbols in the template as variable names (and raising an error if they're not), we could use a syntax like `(%var_name)` to indicate expansion.

I'll copy/paste the example above for easier comparison.

**Before**
```
(macro append_macros ($macro_definitions*)
    (annotate
        (literal $ion_encoding)
        (make_sexp [
            (literal (symbol_table $ion_encoding)),
            (make_sexp [
                (literal macro_table),
                (literal $ion_encoding),
                $macro_definitions
            ])
        ])
    )
)
```
**After**
```
(macro append_macros (macro_definitions*)
    $ion_encoding::(
        (symbol_table $ion_encoding)
        (macro_table $ion_encoding (%macro_definitions))
    )
)
```

As a nice side effect, it means symbol values no longer behave differently from field names and annotations, which has been another stumbling block for newcomers.

Combined, these two changes would eliminate the need for `literal` altogether. To write an s-expression that starts with `.` or `%`, you can use the `.values` macro:
```
((.values %) some_symbol)
((.values .) some_symbol)
```

**Comment by popematt** [comment]
Resolved in #354
